### PR TITLE
Add name propagation for at::alias, add tensor.set_names

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #24203 Implement name inference for t(), transpose(...)
* #24087 Name inference for softmax, log_softmax and Dimname overloads.
* #24028 Fix out= function semantics for named tensors.
* #23973 Update tensor.view_names / tensor.names_ API
* #23962 Rename set_names -> view_names, set_names_ -> names_
* #23804 Implement tensor.align_to(names), torch.align_tensors(t1, t2, ...)
* #24109 Named inference for contiguous(), bernoulli variants, and dropout.
* #24108 Implement name inference rule for empty_like, clone
* #24107 Add `names` argument to ones, rand, randn, zeros, full; fix empty
* **#24202 Add name propagation for at::alias, add tensor.set_names**

tensor.set_names(names) is the out-of-place variant of
tensor.set_names_(names). This naming is probably confusing so I am
taking any and all suggestions.

Test Plan:
- run tests [namedtensor ci]

Differential Revision: [D16773014](https://our.internmc.facebook.com/intern/diff/D16773014)

### CODE REVIEW

**Comment by zou3519** [comment]
This is a resubmit of https://github.com/pytorch/pytorch/pull/23624

**Review by nairbv — APPROVED** [review]
